test/unit_test/data/data_source/weather_data_source_test.dart

```diff
@@ -0,0 +1,108 @@
+// ignore_for_file: prefer_function_declarations_over_variables
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter_training/data/app_exception.dart';
+import 'package:flutter_training/data/data_source/weather_data_source.dart';
+import 'package:flutter_training/data/models/weather/weather.dart';
+import 'package:flutter_training/data/models/weather/weather_request.dart';
+import 'package:flutter_training/views/constants/strings.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:yumemi_weather/yumemi_weather.dart';
+
+import 'weather_data_source_test.mocks.dart';
+
+@GenerateMocks([YumemiWeather])
+void main() {
+  const defaultRequest = WeatherRequest();
+  final client = MockYumemiWeather();
+  final dataSource = WeatherDataSource(client);
+  group(
+    'getWeather',
+    () {
+      test(
+        '''
+           On success of fetchWeather:
+           return Type of Weather
+           ''',
+        () {
+          when(
+            client.fetchWeather(any),
+          ).thenReturn(
+            '''
+            {
+              "weather_condition":"cloudy",
+              "max_temperature":25,"min_temperature":7,
+              "date":"2020-04-01T12:00:00+09:00"
+            }
+            ''',
+          );
+          final act = dataSource.getWeather(request: defaultRequest);
+          final expected = isA<Weather>();
+
+          expect(
+            act,
+            expected,
+          );
+        },
+      );
+
+      test(
+        '''
+        When an exception occurs in fetchWeather:
+        return AppException.unknownError.
```

Comment:
[nits]
`return` より `throw` が適していそうです。
他の箇所も修正お願いします！

```suggestion
        throw AppException.unknownError.
```